terraform/resource_mode.go

```diff
@@ -0,0 +1,9 @@
+package terraform
+
+//go:generate stringer -type=ResourceMode -output=resource_mode_string.go resource_mode.go
+type ResourceMode int
```

Comment:
I think it would be helpful to have a comment here stating that `ResourceMode` is deprecated and to use `addrs.ResourceMode` instead, just in case someone browsing the codebase happens to find this one first or sees both and isn't sure which one to use for new code. What do you think?

(Also, better to put that `go:generate` comment in a different place so godoc won't think it's part of the documentation...)

Reply:
seems reasonable to me!